### PR TITLE
Enable RedisStreamSource scale subresource 

### PR DESCRIPTION
**Pull request by aavarghese**
Fixes https://github.com/knative-sandbox/eventing-redis/issues/38

## Proposed Changes

- Similar to https://github.com/knative-sandbox/eventing-kafka/pull/138
- Related to https://github.com/knative-sandbox/eventing-autoscaler-keda/pull/53

**Release Note**

```
🎁 The RedisStreamSource scale subresource can now be used to scale up and down the underlying deployment
```

**Docs**

### CODE REVIEW

File: source/pkg/apis/sources/v1alpha1/redisstream_types.go [hunk 1]
```diff
@@ -70,6 +70,14 @@ type RedisStreamSourceSpec struct {
 	// deleted when this source is deleted.
 	// +optional
 	Group string `json:"group,omitempty"`
+
+	// Number of desired consumers running in the consumer group. Defaults to 1.
+	//
+	// This is a pointer to distinguish between explicit
+	// zero and not specified.
+	// For round-tripping only.
```
Comment (lionelvillard) [inline source/pkg/apis/sources/v1alpha1/redisstream_types.go:78]: nit: remove `For round-tripping only.`

File: source/pkg/apis/sources/v1alpha1/redisstream_types.go [hunk 2]
```diff
@@ -125,6 +133,11 @@ type RedisStreamSourceStatus struct {
 	// * SinkURI - the current active sink URI that has been configured for the
 	//   Source.
 	duckv1.SourceStatus `json:",inline"`
+
+	// Total number of consumers actually running in the consumer group.
+	// +optional
+	// For round-tripping only.
```
Comment (lionelvillard) [inline source/pkg/apis/sources/v1alpha1/redisstream_types.go:139]: nit: remove `For round-tripping only.`

**Comment by lionelvillard** [comment]
nice!
/lgtm
/approve
/hold

if you want to fix the nits.

**Comment by lionelvillard** [comment]
Can you add a release note?

File: source/pkg/apis/sources/v1alpha1/redisstream_types.go (hunk 2 above)
Comment (aavarghese) [reply source/pkg/apis/sources/v1alpha1/redisstream_types.go]: Done.

**Comment by aavarghese** [comment]
> Can you add a release note?

@lionelvillard added. Thanks for the review!!!

**Comment by aavarghese** [comment]
@lionelvillard could you approve please? knative-sandbox/eventing-autoscaler-keda#53 (depends on this PR) was merged today.

**Pull request merged**